File: Application/Services/Operations/StockProduct/Dtos/ProductKind/ManufacturerDto.cs
using System.Collections.Generic;
using Application.Services.Shared.Dtos;



namespace Application.Services.Operations.StockProduct.ProductKind
{
    public class ManufacturerDto: RootBaseDto
    {
        public string Name { get; set; }
        public int SegmentId { get; set; }
        public SegmentDto Segment { get; set; }
        public List<ModelDto> Models { get; set; }
        public List<ProductDto> Products { get; set; }

    }
}

File: Views/src/components/stock-product/product/dtos/manufacturer-dto.ts
import { RootBase } from "src/shared/entities-dtos/root-base";
import { SegmentDto } from "./segment-dto";
import { ModelDto } from "./model-dto";
import { ProductDto } from "./product";

export class ManufacturerDto extends RootBase {
    name: string;
    segmentId: number;
    segment: SegmentDto;
    models: ModelDto[];
    products: ProductDto[];
}
